site/react/useSmartAccountClient.md

```diff
@@ -0,0 +1,98 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: useSmartAccountClient
+  - - meta
+    - name: description
+      content: An overview of the useSmartAccountClient hook
+  - - meta
+    - property: og:description
+      content: An overview of the useSmartAccountClient hook
+  - - meta
+    - name: twitter:title
+      content: useSmartAccountClient
+  - - meta
+    - name: twitter:description
+      content: An overview of the useSmartAccountClient hook
+---
+
+# useSmartAccountClient
+
+The `useSmartAccountClient` hook is used to create a new [`AlchemySmartAccountClient`](/packages/aa-alchemy/smart-account-client/index) attached to either a `LightAccount` or `MultiOwnerModularAccount` contract using the `AlchemySigner`.
+
+## Import
+
+```ts
+import { useSmartAccountClient } from "@alchemy/aa-alchemy/react";
+```
+
+## Usage
+
+<<< @/snippets/react/useSmartAccountClient.tsx
+
+## Parameters
+
+```ts
+import { type UseSmartAccountClientProps } from "@alchemy/aa-alchemy/react";
+```
+
+### type
+
+`"LightAccount" | "MultiOwnerModularAccount"`
+
+The underlying account type you want to use
+
+### accountParams
+
+```ts
+  | Omit<CreateLightAccountParams, "signer" | "transport" | "chain">
+  | Omit<CreateMultiOwnerModularAccountParams,"signer" | "transport" | "chain">
+  | undefined
+```
+
+An optional param object based on the `type` property passed in above. Allows for overriding the default account parameters.
+
+::: details CreateLightAccountParams
+<<< @/../packages/accounts/src/light-account/account.ts#CreateLightAccountParams
+:::
+
+::: details CreateMultiOwnerModularAccountParams
+<<< @/../packages/accounts/src/msca/account/multiOwnerAccount.ts#CreateMultiOwnerModularAccountParams
+:::
+
+### ...rest
+
+```ts
+Omit<
+  AlchemySmartAccountClientConfig<
+    TTransport,
+    TChain,
+    SupportedAccount<TAccount>
+  >,
+  "rpcUrl" | "chain" | "apiKey" | "jwt" | "account"
+>;
+```
+
+The remaining parameters that are accepted allow for overriding certain properties of the `AlchemySmartAccountClient`
+
+::: details AlchemySmartAccountClientConfig
+<<< @/../packages/alchemy/src/client/smartAccountClient.ts#AlchemySmartAccountClientConfig
+:::
+
+## Return Type
+
+```ts
+import { type UseSmartAccountClientResult } from "@alchemy/aa-alchemy/react";
```

Comment:
should we link to the code itself?

Reply:
yea we sorta do below right? I think this pretty useful if someone wanted to pull this in and use intellisense to view the types

Reply:
oh wait, I lied... this one does not import the code. It's pretty light weight, but I think what would be good is linking to the `AlchemySmartAccountClient` docs below. lemme do that

Reply:
I added a link to smart account client docs